qutip/tests/test_fileio.py

```diff
@@ -44,11 +45,18 @@ def test_separator_detection(self, separator):
         return self.case(_random_file_name(), kwargs)
 
 
-def test_qsave_qload():
+@pytest.mark.parametrize('use_path', [True, False], ids=['Path', 'str'])
+@pytest.mark.parametrize('include_suffix', [True, False],
+                         ids=['with_suffix', 'no_suffix'])
+def test_qsave_qload(use_path, include_suffix):
     ops_in = [qutip.sigmax(),
               qutip.num(_dimension),
               qutip.coherent_dm(_dimension, 1j)]
     filename = _random_file_name()
+    if include_suffix:
```

Comment:
If we remove the if statement that checks for `.qu` as I suggested above, then we can also delete the extra test case here. Although perhaps testing a few suffixes is not a bad idea. E.g. change `'include_suffix', [True, False]` to `'suffix', ['', '.qu', '.dat']`.

Reply:
Testing for different suffixes is a good idea, the code can then be reused later in v5.